File: javascript/backend/samples/content-insights/CloudContentInsightsService.js
const {
    CloudLanguageWeaverClient,
    CloudFileContentInsightsRequest,
    CloudContentInsightsRequest,
} = require("@language-weaver/lw-sdk-js");
const path = require("path");

const getContentInsights = async () => {
    try {
        const cloudLanguageWeaverClient = await new CloudLanguageWeaverClient().build();
        const cloudFileContentInsightsRequest = new CloudFileContentInsightsRequest();
        cloudFileContentInsightsRequest.sourceLanguage = "eng";
        cloudFileContentInsightsRequest.files = path.resolve("resources/input/input1.txt");

        const contentInsightsResult = await cloudLanguageWeaverClient.getCloudContentInsights(cloudFileContentInsightsRequest);
        console.log(contentInsightsResult);
    } catch (e) {
        console.log(e);
    }
}

const getContentInsightsForTranslations = async () => {
    try {
        const cloudLanguageWeaverClient = await new CloudLanguageWeaverClient().build();
        const cloudContentInsightsRequest = new CloudContentInsightsRequest();
        cloudContentInsightsRequest.addTranslationId("translationId");

        const contentInsightsResult = await cloudLanguageWeaverClient.getCloudContentInsightsForTranslations(cloudContentInsightsRequest);
        console.log(contentInsightsResult);
    } catch (e) {
        console.log(e);
    }
}

getContentInsights();
getContentInsightsForTranslations();

File: dotnet/Samples/ContentInsights/CloudContentInsightsService.cs
using LanguageWeaver.Sdk.Common.Cloud;
using LanguageWeaver.Sdk.Constants;
using LanguageWeaver.Sdk.ContentInsights.Cloud.Request;
using LanguageWeaver.Sdk.ContentInsights.Cloud.Result;
using LanguageWeaver.Sdk.Translate.Cloud.Request;

namespace LanguageWeaver.Sdk.Samples.ContentInsights;

public class CloudContentInsightsService
{
    public static void Main()
    {
        using var lwClient = new CloudLanguageWeaverClient().Build();
        CreateContentInsights(lwClient);
        CreateContentInsightsForExistingTranslations(lwClient);
    }

    private static void CreateContentInsights(CloudLanguageWeaverClient lwClient)
    {
        CloudFileContentInsightsRequest cloudContentInsightsRequest = new CloudFileContentInsightsRequest()
        {
            SourceLanguage = "eng",
            FilePath = Path.Combine(Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName,
                "Resources", "Input", "input1.txt")
        };

        CloudContentInsightsResult cloudContentInsightsResult =
            lwClient.GetContentInsights(cloudContentInsightsRequest);
        // handle result
    }

    private static void CreateContentInsightsForExistingTranslations(CloudLanguageWeaverClient lwClient)
    {
        CloudTranslateFileRequest translateFileRequest = new CloudTranslateFileRequest()
        {
            SourceLanguageId = "eng",
            TargetLanguageId = "fra",
            Model = "generic",
            InputFormat = Format.Plain,
            InputFile = Path.Combine(Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName,
                "Resources", "Input", "input1.txt"),
            OutputFile = Path.Combine(Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName,
                "Resources", "Output", "input1-translated.txt")
        };

        var cloudTranslationFileResult = lwClient.TranslateFile(translateFileRequest);
        CloudContentInsightsRequest contentInsightsForTranslationsRequest =
            new CloudContentInsightsRequest().AddTranslationId(cloudTranslationFileResult.TranslationId);

        CloudContentInsightsResult cloudContentInsightsResult =
            lwClient.GetContentInsightsForTranslations(contentInsightsForTranslationsRequest);
        // handle result
    }
}